src/fsource/lexer.py

```diff
@@ -146,8 +146,9 @@ def get_lexer_regex():
                          | G      \d+ (?: \.\d+  (?: E\d+)?)?
                          | L      \d+
                          | A      \d*
-                         | [XP]
-                         )(?=\s*[:/,)])"""
+                         | X
+                         )(?=\s*[:/,)])
+                    |(?:-?\d+P)(?=\s*,?[\d:DEFG])"""
```

Comment:
Could you format this a bit nicer?  It's hard to read this way.

Also, I think the `-` is unnecessary here.

Also, are you sure `P` always needs to be followed by `DEFG`? Can it not be terminated there?

Reply:
I've seen reference to negative scale factors but you're right; the `\d` should be enough; the `-?` doesn't add anything.

I believe `P` only affects real formats which is why I originally set `DEFG` as a lookahead assertion. It's not relevant if the `P` is followed by a comma, but in the case of no comma something like `1PA8` doesn't make much sense. It might be reasonable to remove the whole `[\d:DEFG]` pattern. I doubt it will make much difference in practice.

I'll make those changes and reformat the regex for readability later today.